SmartDeviceLink/SDLSoftButtonManager.h

```diff
@@ -0,0 +1,68 @@
+//
+//  SDLSoftButtonManager.h
+//  SmartDeviceLink
+//
+//  Created by Joel Fischer on 2/22/18.
+//  Copyright © 2018 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@protocol SDLConnectionManagerType;
+
+@class SDLFileManager;
+@class SDLSoftButtonObject;
+@class SDLSoftButtonState;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ The handler run when the update has completed
+
+ @param error An error if the update failed and an error occurred
+ */
+typedef void(^SDLSoftButtonUpdateCompletionHandler)(NSError *__nullable error);
+
+@interface SDLSoftButtonManager : NSObject
+
+/**
+ HAX: This is necessary due to a Ford Sync 3 bug that doesn't like Show requests without a main field being set (it will accept them, but with a GENERIC_ERROR, and 10-15 seconds late...)
+ */
+@property (copy, nonatomic, nullable) NSString *currentMainField1;
+
```

Comment:
This HAX doesn't seem to work with _Manticore_ when all 4 of the text-fields are set. The`textAndGraphicManager` update gets sent first and all the text-fields get set correctly. Then the `softButtonManager` update gets sent next and overwrites the `mainField2` and `mainField3` with an empty string. The result is text flashes for a fraction of a second in `mainField2` and `mainField3`.

Reply:
I'm not sure what's causing this for you, but the SoftButtonManager does not overwrite anything but `mainField1`. It's `mainField2` and 3 are nil, which does not change the current text. I've tested it and it seems fine.

Reply:
I see the bug now, trying to track it down

Reply:
Generic HMI issue filed: https://github.com/smartdevicelink/generic_hmi/issues/62